modules/network/pom.xml

```diff
@@ -48,6 +48,11 @@
             <artifactId>ignite-core</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.ignite</groupId>
+            <artifactId>ignite-network-messages</artifactId>
```

Comment:
If we add ignite-network-messages like this, annotation processor will make its way to the client's code. I think annotation processor should be a separate module that is only added as a compile time dependency. Unfortunately, this means that APIs like NetworkMessage should be moved to yet another module (network-api?)
In my opinion the dependency tree should look like this:

Network -> (compile) NetworkMessage -> NetworkApi
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-> NetworkApi

Reply:
How is that going to help? "Compile" scope is transitive, so this processor will be visible to the client code in your approach as well. Or am I missing something here?

Reply:
Sorry, what I meant was "optional", not "compile"